src/Abi/Encode.elm

```diff
@@ -746,5 +746,6 @@ tillMod64 n =
 stringToHex : String -> String
 stringToHex strVal =
     UTF8.toBytes strVal
-        |> List.map Hex.toString
+        |> List.map
+            (Hex.toString >> String.padLeft 2 '0')
```

Comment:
I found this when the newline character `0x0A` would cause garbled string encoding. I found that this line converts `0x0A` into the **single-char-length** string `a`, which means that a string like 'hi\nhi' will result in a hex string of `6869a6869` (note the odd-numbered length) rather than `68690a6869`.

As a funny side-note, I tried to use [this online hex converter](https://codebeautify.org/string-hex-converter) to get the example hex values above... and it has the same bug! Try converting
```
hi
hi
```
to hex, then take the outputted value and try to convert it back to text... and you'll see the exact garbledness that I was running into.

Reply:
Ah, totally. Good catch.

I'm wrong about the empty string producing something of length 2 as well. Reread the code, and saw the error in my reasoning.